### DISCUSSION TITLE
Plugins compatibility after upgrading from td-agent v4 to fluentd-package v5

### DISCUSSION THREAD

**Discussion by shubher13** [fluent/fluentd#4322] [Q&A]
I am using td-agent v4.5.1 with bunch of additional td-agent based plugins and planning to upgrade to fluentd-package v5.
So would all those plugins continue to work with v5 after ruby version upgrade from 2.7.8 -> 3.2.2 in the fluentd-package?

**Comment by kenhys** [comment]
Maybe most of plugins will work, but not sure.

If you want to ensure it for bunch of additional plugins, you can check like this:

1. list bunch of additional td-agent based plugins with version information 
2. access rubygems via API

For example, if you want check latest version, you can do it with:

```
 curl --silent https://rubygems.org/api/v1/versions/fluent-plugin-elasticsearch/latest.json | jq --raw-output ".version"
5.3.0
```

Then fetch required ruby version like this:

```
curl --silent https://rubygems.org/api/v2/rubygems/fluent-plugin-elasticsearch/versions/5.3.0 |jq --raw-output ".ruby_version" 
>= 2.3
```

Thus you can know that fluent-plugin-elasticsearch work with ruby 2.3 or later.

If all of additional plugins pass above check, you can safely migrate to fluent-package.
